tests/DCPS/Restart/RestartTest.cpp

```diff
@@ -74,28 +124,55 @@ struct Application {
       TheServiceParticipant->shutdown();
     }
     ACE::fini();
+    std::cout << "Application " << id_ << " Ended\n";
   }
 
   static int instances_;
+  static int total_instances_;
 
+  int id_;
   std::string transport_config_name_;
   DDS::DomainParticipant_var participant_;
 };
 
 int Application::instances_ = 0;
+int Application::total_instances_ = 0;
 
-int main()
+int main(int argc, char* argv[])
 {
-  {
-    Application a1;
-    {
-      Application a2;
+  const std::string usage =
+    std::string("usage: ") + argv[0] + " [--help|-h] [--secure]\n";
+  for (int i = 1; i < argc; i++) {
+    const std::string argument(argv[i]);
+    if (argument == "--help" || argument == "-h") {
+      std::cout << usage;
+      return EXIT_SUCCESS;
+    } else if (argument == "--secure") {
+#ifdef OPENDDS_SECURITY
+      TheServiceParticipant->set_security(true);
+#endif
     }
+  }
+
+  try {
     {
-      Application a3;
+      Application a1;
+      ACE_OS::sleep(1);
+      {
+        Application a2;
+        ACE_OS::sleep(1);
+      }
+      {
+        Application a3;
+        ACE_OS::sleep(1);
+      }
     }
+    Application a4;
+    ACE_OS::sleep(1);
+  } catch (std::string error) {
+    ACE_ERROR((LM_ERROR, "Caught error: %C\n", error.c_str()));
```

Comment:
```suggestion
    ACE_ERROR((LM_ERROR, "Caught Error: %C\n", error.c_str()));
```
(scoreboard will highlight it)

Reply:
It highlights "Error", but not "error"?